Date: Mar 10, 2015 10:41 PM

From: 177d90643ebf95c28b3d0516d110856ed853fac6 Mar 10, 2015 10:57 PM

Subject: [PATCH] RDKSYSINTSW-1476: Update CCSP recipes for RDK-B

	merge Cisco's 2015-01-23 Belvedere drop.
	Adding 'libdir' from the .la file to our library search paths
	breaks crosscompilation horribly.  We cheat here and don't add
	it, instead adding the path where we found the .la.  -CL

Source: COMCAST


Upstream-Status: Pending

Signed-off-by: Steve Maynard <steve.maynard@twcable.com>
---

--- a/build-aux/ltmain.sh	2013-02-19 17:38:50.831930000 +0800
+++ b/build-aux/ltmain.sh	2013-02-19 17:58:15.643132000 +0800
@@ -6504,8 +6504,14 @@
 	    absdir="$abs_ladir"
 	    libdir="$abs_ladir"
 	  else
-	    dir="$lt_sysroot$libdir"
-	    absdir="$lt_sysroot$libdir"
+	    # Adding 'libdir' from the .la file to our library search paths
+	    # breaks crosscompilation horribly.  We cheat here and don't add
+	    # it, instead adding the path where we found the .la.  -CL
+	    dir="$abs_ladir"
+	    absdir="$abs_ladir"
+	    libdir="$abs_ladir"
+	    #dir="$lt_sysroot$libdir"
+	    #absdir="$lt_sysroot$libdir"
 	  fi
 	  test "X$hardcode_automatic" = Xyes && avoidtemprpath=yes
 	else
@@ -6904,7 +6910,7 @@
 	      fi
 	    else
 	      # We cannot seem to hardcode it, guess we'll fake it.
-	      add_dir="-L$libdir"
+	      #add_dir="-L$libdir"
 	      # Try looking first in the location we're being installed to.
 	      if test -n "$inst_prefix_dir"; then
 		case $libdir in
@@ -7059,7 +7065,18 @@
 		  fi
 		  ;;
 		*)
-		  path="-L$absdir/$objdir"
+		  #path="-L$absdir/$objdir"
+                    # OE sets installed=no in staging. We need to look in $objdir and $absdir,
+                   # preferring $objdir. RP 31/04/2008
+                  if test -f "$absdir/$objdir/$depdepl" ; then
+		    depdepl="$absdir/$objdir/$depdepl"
+		    path="-L$absdir/$objdir"
+                  elif test -f "$absdir/$depdepl" ; then
+		    depdepl="$absdir/$depdepl"
+		    path="-L$absdir"
+                  else
+		    path="-L$absdir/$objdir"
+                  fi
 		  ;;
 		esac
 		else
